### DISCUSSION TITLE
vweb: create an extensive standalone documentation

### DISCUSSION THREAD

**Discussion by Zerotask** [vlang/v#11823] [Ideas]
Since the vweb web framework will get bigger and bigger, it needs a standalone documentation page. For the beginning there could be a **doc/vweb.md** file with a clear structure, where all the important points are explainined. This will allow people to create good Web APIs / websites.

At the moment there are just references to other projects or code samples but there is no single complete documentation about it. This makes it harder to jump on the train.

**Comment by JalonSolov** [comment]
You are absolutely correct, and also _far_ too early for this.  vweb is still not considered complete, and is currently undergoing a major refactoring of the code.

I agree that this needs to be done, but I don't think it's a high priority right now.

Of course you are welcome to start on such documentation if you like, just be prepared to change some or all of it as vweb matures.

**Comment by Zerotask** [comment]
I don't think it's too early. Web programming is a major topic for many people including me and the current framework has a solid foundation. It should have a better documentation.
If it gets refactored, then you simply update the documentation.

**Comment by canadaduane** [comment]
Three insights on documentation that might be helpful here, taken from various communities:

1. A (short) synopsis of the [architecture](https://matklad.github.io/2021/02/06/ARCHITECTURE.md.html) can be helpful, even if the details aren't documented. This doesn't have to take a lot of time, and if short, can be changed easily as the refactor takes place.

2. Documentation has [four quadrants](https://documentation.divio.com/): practical<->theoretical and studying<->working. In the early stages of a project, it can be useful to focus on the "Discussions" (theoretical/studying) quadrant in order to help others understand, but without the burden of writing or maintaining tutorials, references, or how-to guides.

3. "[How Docs vs. Why Docs](https://codeascraft.com/2018/10/10/etsys-experiment-with-immutable-documentation/)". The key insight here is that "why" is slow to change, whereas "how" is an evolving target.

**Comment by nedpals** [comment]
I think there's no reason to close this issue. This is an important one to keep track and this should be closed once we have it.
